lib/builder.js

```diff
@@ -111,7 +112,8 @@ Builder.prototype.build = function (willReadStringTree) {
       name: getDescription(tree),
       broccoliNode: true,
       broccoliId: node.id,
-      broccoliCachedNode: false
+      broccoliCachedNode: false,
+      broccoliPluginName: getPluginName(tree)
```

Comment:
@rwjblue i think leaving them as discrete parts is fine.  I was thinking mainly of broccolil-viz, since what it shows is a name, but upon reflection it's less of an issue there since

a) you have more context as you're already looking at the surrounding graph and
b) it seems fine to just have broccoli viz learn about certain kinds of nodes  and display certain fields if they're present (such as broccoliAnnotation).